src/udiff/_vjp_diffs.py

```diff
@@ -553,12 +553,12 @@ def matmul_adjoint_1(A, G, A_ndim, B_meta):
 
 def matmul_vjp_0(ans, A, B):
     A_meta = metadata(A)
-    B_ndim = np.ndim(B)
+    B_ndim = B.ndim
     return lambda g: matmul_adjoint_0(B, g, A_meta, B_ndim)
 
 
 def matmul_vjp_1(ans, A, B):
-    A_ndim = np.ndim(A)
+    A_ndim = A.ndim
```

Comment:
Same here.

Reply:
I think there is no difference between these two lines. Is it better to use `np.ndim`?

Reply:
If you’re sure this is a `DiffArray`, then it’s the same, otherwise the other package might not have implemented `.ndim` in which case the `np.ndim` will always do the right thing.